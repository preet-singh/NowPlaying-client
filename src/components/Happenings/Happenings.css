.Happenings {
    margin-top: 20px;
    flex: 1;
    background-color: rgba(0,0,0,0.8);
    color: white;
}
    .Happenings h3 {
        font-size: 28px;
        background-color: black;
    }

    .happening-label {
        font-weight: bold;
    }

.Happenings li:hover {
    background-color: #00d838;
    color: black;
}

.Happenings li:hover a {
    color: black;
}


@media screen and (min-width: 768px){

    .Happenings h3 {
        padding: 15px;
    }

    .Happenings ul {
        list-style-type: none;
        /* display: flex;
        flex-direction: row;
        justify-content: space-evenly; */
        width: 60%;
        min-height: 200px;
        border: 1px solid grey;
        transform: translateX(35%);

    }

    .Happenings ul li {
        list-style-type: none;
        /* background-color: yellow; */
        /* border: 1px solid yellow; */
        /* height: 175px; */
        /* width: 30%; */
        text-align: center;
        justify-content: center;
        padding-top: 8px;
        padding-bottom: 8px;
    }

}