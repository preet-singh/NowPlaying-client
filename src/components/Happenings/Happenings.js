import React, { Component } from 'react';
import UserContext from '../../utils/context';
import AuthApiService from '../../utils/auth-service';
import './Happenings.css'

class Happenings extends Component {
  static contextType = UserContext;

  constructor(props){
    super(props);

    this.state = {
      error: null,
      happenings: [],
      isLoading: false
    };
  }

  componentDidMount() {
    AuthApiService.getHappeningEvents()
      .then(data => {
        this.context.setHappenings(data);
        this.setState({
          happenings: data
        })
      })
      .catch(res => {
        this.setState({
          error: res.error
        })
      })
    }

  render() {
    let firstHappenings = this.state.happenings.slice(0,this.state.happenings.length - 1);
    let lastHappening = this.state.happenings[this.state.happenings.length - 1] || {};
    return (
      <div className="Happenings">
        <h3>What's happening</h3>
        <ul className="happenings-list">
          {firstHappenings.map(event =>
              (event.media_title)
               ? <>
                  <li className="thread-created">A thread for the {event.media_type.slice(0,event.media_type.length - 1)}, <span className="media-title">{event.media_title}</span>, was created</li>
                  <hr className="underline" />
                 </>
               : 
                <>
                 <li className="comment-created">{event.username} commented <span className="user-comment">"{event.user_comment}"</span> in the <span className="media-title">{event.media_title_comments}</span> thread</li>
                 <hr className="underline" />
                </>
          )}
          {(lastHappening.media_title)
              ? 
                <li className="thread-created">A thread for the {lastHappening.media_type.slice(0,lastHappening.media_type.length - 1)}, <span className="media-title">{lastHappening.media_title}</span>, was created</li>
              : 
                <li className="comment-created">{lastHappening.username} commented <span className="user-comment">"{lastHappening.user_comment}"</span> in the <span className="media-title">{lastHappening.media_title_comments}</span> thread</li> 
          }
        </ul>
      </div>
    );
  }
}

export default Happenings;