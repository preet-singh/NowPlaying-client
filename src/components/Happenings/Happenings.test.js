import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import Happenings from './Happenings';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json'

describe('Happenings testing', () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <Happenings />
            </BrowserRouter>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the hapenning table', () => {
        const wrapper = shallow(<Happenings />)
        expect(toJson(wrapper)).toMatchSnapshot()
    });

});