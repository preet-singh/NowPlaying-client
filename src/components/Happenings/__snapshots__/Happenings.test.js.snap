// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Happenings testing renders the hapenning table 1`] = `
<div
  className="Happenings"
>
  <h3>
    ACTIVITY
  </h3>
  <ul
    className="happenings-list"
  />
</div>
`;
